functionary/core/models/function.py

```diff
@@ -80,9 +91,23 @@ def deactivate(self):
         """Deactivate the function and pause any associated scheduled tasks"""
         self.active = False
         self.save()
+        self.update_scheduled_tasks(enable=False)
 
+    def update_scheduled_tasks(self, enable: bool) -> None:
```

Comment:
The two different behaviors here feels like belong in two separate functions.  But also, I don't think calling this with enable = True would ever be a good idea, so I'd probably just eliminate that path and make this `pause_scheduled_tasks`.